base/special/hyperbolic.jl

```diff
@@ -260,20 +256,13 @@ function atanh(x::T) where T <: Union{Float32, Float64}
     if absx > 1
         atanh_domain_error(x)
     end
-    if absx < T(2)^-28
-        # in a)
-        return x
-    end
     if absx < T(0.5)
-        # in b)
+        # in a)
         t = absx+absx
-        t = T(0.5)*log1p(t+t*absx/(T(1)-absx))
-    elseif absx < T(1)
-        # in c)
-        t = T(0.5)*log1p((absx + absx)/(T(1)-absx))
-    elseif absx == T(1)
-        # in d)
-        return copysign(T(Inf), x)
+        t = log1p(t+t*absx/(T(1)-absx))
```

Comment:
would fma help here?

Reply:
It actually doesn't. It raises the error from 1.44 to 1.46 ULP. I have no idea why though.

Reply:
That's kind of bizarre.

Reply:
I actually think it doesn't have any accuracy impact. The quick testing I'm doing is somewhat sensitive to the chosen points.

Reply:
The new version of this I just pushed is `log1p(2x/(1-x))` which is a minor simplification with no accuracy impact